### ISSUE TITLE
How to inject a repository into another one ?

### ISSUE THREAD

**Issue by booradleys** [typeorm/typeorm-typedi-extensions#17]
My repository candidatRepository is undefined.
How can i inject it into another repository ?
````
import {Service} from "typedi";
import { Repository, EntityRepository } from "typeorm";
import {OrmCustomRepository} from "../../node_modules/typeorm-typedi-extensions/decorators/OrmCustomRepository";
import {TCandidatRepository} from "../repository/TCandidatRepository";
import {TCandidatStats} from "../entity/TCandidatStats";

// create custom Repository class
@Service()
@EntityRepository(TCandidatStats)
export class TCandidatStatsRepository extends Repository<TCandidatStats>
{
    //property injection
    @OrmCustomRepository(TCandidatRepository)
    private candidatRepository: TCandidatRepository

    updateDateConnexion(candidat_id: number) {
            let self = this;
            return  this.findOneById(candidat_id)
                    .then(function(candidatStats) {
                        if(!candidatStats) {
                            return self.candidatRepository.findOneById(candidat_id) 
// =>  self.candidatRepository IS UNDEFINED
                            .then(function(candidat) {
                                console.log("SECOND");
                                candidatStats = new TCandidatStats();
                                candidatStats.setCandidat(candidat);
                                candidatStats.setDateInscription(new Date());
                                return candidatStats;
                            });
                        }
                        return candidatStats;
                    })
                    .then(function(candidatStats) {
                        candidatStats.setDateDerniereConnection(new Date());
                        return self.persist(candidatStats);
                    });
            
    }	
}
````

**Comment by 19majkel94** [comment]
**Protip**: Try to use lambda syntax which has lexicaly scoped this to avoid weird binding to self 😉 

Do you use old `typeorm-typedi-extensions` version? I see `import {OrmCustomRepository} from "../../node_modules/typeorm-typedi-extensions/decorators/OrmCustomRepository";` which has been patched long long time ago.

**Comment by booradleys** [comment]
@19majkel94 , yes i'm using OrmCustomRepository everywhere in my controllers and services mainly.
If typeorm-typedi-extensions is deprecated,
Do i have to replace :
````
 @OrmCustomRepository(TCandidatRepository)
    private candidatRepository: TCandidatRepository
````
by :
````
let candidatRepository = typeorm.getConnection().getCustomRepository(TCandidatRepository);
````
?

**Comment by 19majkel94** [comment]
No 😆 Old version = 0.0.9. There were no exported `OrmCustomRepository` decorator and this has been fixed in 0.0.10: https://github.com/typeorm/typeorm-typedi-extensions/commit/e0afec4a2b48578cd9f22c07353ef5ead3cc16dc

I don't know why `candidatRepository IS UNDEFINED` but I think that repository injected into repository is an anti pattern. You have something wrong with your data models, the relations are badly designed if you need this kind of dirty hacks.

**Comment by booradleys** [comment]
Ah ok thank you!
Now i'm using
import { OrmCustomRepository } from "typeorm-typedi-extensions";

I'm going to check my entities.
Thank you @19majkel94
